test/server/cards/02_SHD/events/PalpatinesReturn.spec.ts

```diff
@@ -0,0 +1,55 @@
+describe('Palpatine\'s Return', function() {
+    integration(function(contextRef) {
+        describe('Palpatine\'s Return\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        base: 'dagobah-swamp',
+                        leader: 'grand-moff-tarkin#oversector-governor',
+                        hand: ['palpatines-return'],
+                        discard: ['supreme-leader-snoke#shadow-ruler', 'avenger#hunting-star-destroyer', 'waylay', 'foundling'],
+                        resources: 21
+                    },
+                    player2: {
+                        discard: ['wampa']
+                    }
+                });
+            });
+
+            it('allows to play a unit from the discard', function () {
+                const { context } = contextRef;
+                const reset = () => {
+                    context.player1.moveCard(context.palpatinesReturn, 'hand');
+                    context.player2.passAction();
+                };
+
+                // Scenario 1: Play a force unit for 8 less
+                context.player1.clickCard(context.palpatinesReturn);
+                expect(context.player1).toBeAbleToSelectExactly([context.supremeLeaderSnoke, context.avenger]);
+
+                let exhaustedResourcesBeforeAction = context.player1.exhaustedResourceCount;
+                context.player1.clickCard(context.supremeLeaderSnoke);
+                expect(context.player1.exhaustedResourceCount).toBe(exhaustedResourcesBeforeAction);
```

Comment:
Could we play an out-of-aspect unit so we can get the cost above 8 and confirm the exact amount of the discount?